docs/troubleshooting.md

```diff
@@ -99,6 +99,108 @@ for p in $(oc get pods -l component=es -o jsonpath={.items[*].metadata.name}); d
 done
 ```
 
+## Fluentd
+### Fluentd is holding onto deleted journald files that have been rotated
+This [issue](https://bugzilla.redhat.com/show_bug.cgi?id=1664744) causes `/var/log` to fill up with file handles of 
+deleted journald files.  The only known solution at this time is to periodically cycle Fluentd.  An OKD `CronJob` can be added to perform this function, for example, by checking every 10 minutes the amount of available space to fluentd:
+```
+apiVersion: v1
+kind: Template
+metadata:
+  name: fluentd-reaper
+objects:
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: Role
+  metadata:
+    name: fluentd-reaper
+  rules:
+  - apiGroups:
+    - ""
+    resources:
+    - pods
+    verbs:
+    - delete
+  - apiGroups:
+    - ""
+    resources:
+    - pods/exec
+    verbs:
+    - create
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: RoleBinding
+  metadata:
+    name: fluentd-reaper
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: Role
+    name: fluentd-reaper
+  subjects:
+  - kind: ServiceAccount
+    name: aggregated-logging-fluentd
+    namespace: openshift-logging
+- apiVersion: batch/v1beta1
+  kind: CronJob
+  metadata:
+    name: fluentd-reaper
+    labels:
+       provider: openshift
+       logging-infra: fluentd-reaper
+  spec:
+    schedule: "${REAP_SCHEDULE}"
+    jobTemplate:
+      spec:
+        template:
+          metadata:
+            labels:
+              provider: openshift
+              logging-infra: fluentd-reaper
+          spec:
+            serviceAccount: aggregated-logging-fluentd
+            serviceAccountName: aggregated-logging-fluentd
+            containers:
+            - env:
+              - name: ENABLED
+                value: "true"
+              - name: REAP_THRESHOLD
+                value: "${REAP_THRESHOLD}"
+              name: cli
+              image: ${CLI_IMAGE}
+              command: ["/bin/bash", "-c"]
+              args:
+                - if [ "$ENABLED" != "true" ]; then
+                    echo "Job is disabled; exiting early";
+                    exit 0;
+                  fi;
+                  echo "Checking fluentd pods for space issues on /var/log...";
+                  pods=$(oc get pods -l component=fluentd -o jsonpath={.items[*].metadata.name});
+                  for p in $pods; do
+                    echo "Checking $p...";
+                    space=$(oc exec -c fluentd-elasticsearch $p -- bash -c 'df -Ph /var/log | tail -1 | cut -s -d " " -f 12 | rev | cut -b 2- | rev');
+                    echo "Capacity $space";
+                    if [ $space -gt ${REAP_THRESHOLD} ] ; then 
+                      echo "Used capacity exceeds threshold. Deleting $pod";
+                      oc delete pod $p ;
```

Comment:
do you have to wait at all for stability or anything here or is it ok to just kill all the pods if they're all running out of space?

Reply:
Assuming the schedule is not a tight window, deleting the pod is sufficient.  I dont see any reason to wait for anything.  There is not sane shutdown to fluent anyway as best I can tell. I don't belive it responds to any SIGHUP

Reply:
I think SIGHUP just causes it to reload config. If it doesn't cause dropping open files, then it won't solve the problem.